src/main/kotlin/org/rust/lang/core/macros/MacroExpander.kt

```diff
@@ -407,6 +354,61 @@ private class MacroPattern private constructor(
             generateSequence(firstChild) { it.nextSibling }
                 .filter { it !is PsiWhiteSpace && it !is PsiComment }
 
+        fun parse(builder: PsiBuilder, kind: FragmentKind): Boolean {
```

Comment:
Consider moving it to a better place and maybe turning it into extension property of `FragmentKind`.